uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -566,45 +565,70 @@ sub identify_scan_db {
-            print &in_range($yspace, $rowref->{'yspace_range'}) ? "yspace\t" : '';
-            print &in_range($zspace, $rowref->{'zspace_range'}) ? "zspace\t" : '';
-            print &in_range($slice_thickness, $rowref->{'slice_thickness_range'}) ? "ST\t" : '';
-            print &in_range($xstep, $rowref->{'xstep_range'}) ? "xstep\t" : '';
-            print &in_range($ystep, $rowref->{'ystep_range'}) ? "ystep\t" : '';
-            print &in_range($zstep, $rowref->{'zstep_range'}) ? "zstep\t" : '';
-            print &in_range($time, $rowref->{'time_range'}) ? "time\t" : '';
+            if($sd_regex && ($series_description =~ /$sd_regex/i)) {
+                print "series_description\t";
+            }
+            print &in_range($tr,     "$tr_min-$tr_max")         ? "TR\t"     : '';
+            print &in_range($te,     "$te_min-$te_max")         ? "TE\t"     : '';
+            print &in_range($ti,     "$ti_min-$ti_max")         ? "TI\t"     : '';
+            print &in_range($xspace, "$xspace_min-$xspace_max") ? "xspace\t" : '';
+            print &in_range($yspace, "$yspace_min-$yspace_max") ? "yspace\t" : '';
+            print &in_range($zspace, "$zspace_min-$zspace_max") ? "zspace\t" : '';
+            print &in_range($xstep,  "$xstep_min-$xstep_max")   ? "xstep\t"  : '';
+            print &in_range($ystep,  "$ystep_min-$ystep_max")   ? "ystep\t"  : '';
+            print &in_range($zstep,  "$zstep_min-$zstep_max")   ? "zstep\t"  : '';
+            print &in_range($time,   "$time_min-$time_max")     ? "time\t"   : '';
+            print &in_range($slice_thickness, "$slice_thick_min-$slice_thick_max") ? "ST\t" : '';
             print "\n";
         }
-        
-	if ($sd_regex) {
+
+	    if ($sd_regex) {
             if ($series_description =~ /$sd_regex/i) {
                 return &scan_type_id_to_text($rowref->{'Scan_type'}, $dbhr);
             }
-	}
-	else {
-         	if ((!$rowref->{'TR_range'} || &in_range($tr, $rowref->{'TR_range'}))
-                && (!$rowref->{'TE_range'} || &in_range($te, $rowref->{'TE_range'}))
-                && (!$rowref->{'TI_range'} || &in_range($ti, $rowref->{'TI_range'}))
-                && (!$rowref->{'slice_thickness_range'} || &in_range($slice_thickness, $rowref->{'slice_thickness_range'}))
-
-                && (!$rowref->{'xspace_range'} || &in_range($xspace, $rowref->{'xspace_range'}))
-                && (!$rowref->{'yspace_range'} || &in_range($yspace, $rowref->{'yspace_range'}))
-                && (!$rowref->{'zspace_range'} || &in_range($zspace, $rowref->{'zspace_range'}))
-
-                && (!$rowref->{'xstep_range'} || &in_range($xstep, $rowref->{'xstep_range'}))
-                && (!$rowref->{'ystep_range'} || &in_range($ystep, $rowref->{'ystep_range'}))
-                && (!$rowref->{'zstep_range'} || &in_range($zstep, $rowref->{'zstep_range'}))
-                && (!$rowref->{'time_range'} || &in_range($time, $rowref->{'time_range'}))) {
+	    } else {
+         	if (
```

Comment:
Oops..I missed these faulty `if` statements during my first review. These should all be changed from `if(!$x && !$y)` to `if(!defined $x && !defined $y)` (see comment made in function `in_range` for the tests on `$min` and `$max`)

Reply:
done! @nicolasbrossard Ready for review again and testing when you are. Thanks!